source/databricks/calculation-engine/package/calculation_input/table_reader.py

```
@@ -181,8 +181,14 @@ def _fix_metering_point_type(self, df: DataFrame) -> DataFrame:
                 col(Colname.metering_point_type)
                 == InputMeteringPointType.EFFECT_SETTLEMENT.value,
                 lit(MeteringPointType.EFFECT_SETTLEMENT.value),
-            ),
+            )
+            .otherwise(lit("Unknown type")),
         )
+        # The otherwise is to avoid changing the nullability of the column.
+        # Raise error if metering point type is unknown.
+        if df.where(col(Colname.metering_point_type) == "Unknown type").count() > 0:
+            raise ValueError("Unknown metering point type")
```

I think we should avoid the `.count()` as it might not be an entirely inexpensive operation. I think we'd be better of not enforcing a performance penalty in every calculation but rather just let it fail during write.